internal/infrastructure/kubernetes/deployment.go

```diff
@@ -0,0 +1,218 @@
+package kubernetes
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/pointer"
+
+	"github.com/envoyproxy/gateway/internal/ir"
+)
+
+const (
+	// envoyContainerName is the name of the Envoy container.
+	envoyContainerName = "envoy"
+	// envoyNsEnvVar is the name of the contour namespace environment variable.
```

Comment:
Question: should this be `the name of the Envoy Gateway namespace environment variable` instead of `.... countour ....`?